integration/record.sh

```diff
@@ -0,0 +1,336 @@
+#!/bin/env bash
+# Copyright 2023 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Given an active connection to a kubernetes cluster configured with Tekton Pipelines and Tekton
+# Chains, execute a set of PipelineRuns to demonstrate the different ways Chains generates
+# SLSA Provenance.
+#
+# This script has been tested against a kind cluster, but it should technically work against any
+# kubernetes cluster. To setup a kind cluster:
+#
+#   kind create cluster
+#   kubectl apply --filename https://storage.googleapis.com/tekton-releases/pipeline/latest/release.yaml
+#   kubectl apply --filename https://storage.googleapis.com/tekton-releases/chains/latest/release.yaml
+#   cosign generate-key-pair k8s://tekton-chains/signing-secrets
+#
+# The ServiceAccount responsible for running Tasks/Pipelines should have the proper secrets linked
+# so Chains can push contents to the OCI repository. When using a kind cluster, the ServiceAccount
+# used is called "default". Below is one way to just upload your current credentials to the cluster
+# and link to the ServiceAccount:
+#
+#   kubectl -n default create secret docker-registry $USER \
+#     --from-file=.dockerconfigjson=$HOME/.docker/config.json
+#   kubectl patch serviceaccount default -p '{"imagePullSecrets": [{"name": "'$USER'"}]}'
+#   kubectl patch serviceaccount default -p '{"secrets": [{"name": "'$USER'"}]}'
+#
+#
+# Usage: ./integration/record.sh
+#   The environment variable IMAGE_URL must point to an OCI repository, e.g. quay.io/lucarval/spam
+#   You must have write access to the repository.
```

Comment:
Could we spin up a local registry?

Reply:
Ideally, yes. But I don't know how to make that accessible from within the cluster by Chains. Probably not worth chasing that for this PR.